### ISSUE TITLE
'Go to Type Definition' not working if the definition file is a closed .ts file

### ISSUE THREAD

**Issue by two-bridges** [Microsoft/vscode#58759]
Issue Type: <b>Bug</b>

The issue is as follows: 
* Open a typescript file in the editor.  
* Select a function that is defined in another file.  
* Click "Goto Definition".  
* If the code definition resides in file that is not in the list of opened editors, then VSCode does not open the target file.  Instead, it displays an error message (bottom-right toast popup) with the error message "Cannot read property 'isOpened' of undefined"

Notes:
* Note 1: if the code definition resies in a file that IS an open editor, then the error does not occur and the Goto definition succeeds.  
* Note 2:  I tried a full system restart and the problem still persists.  
* Note 3: if I open a PHP code project in VSCode, then the problem does not occur at all and Goto defition works perfectly, as intended.  It appears to be a problem with opening a typescript file.  

I have disabled all extensions and the problem persists.  

I have been coding all day and the problem only just started occurring after a restart.  I assume that an update was to blame.  

VS Code version: Code 1.27.2 (f46c4c469d6e6d8c46f268d1553c5dc4b475840f, 2018-09-12T16:17:45.060Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz (8 x 4008)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|31.69GB (21.41GB free)|
|Process Argv|C:\Users\Dean\AppData\Local\Programs\Microsoft VS Code\Code.exe c:\code\yf\rpt-www|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (39)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-css-formatter|aes|1.0.1
sort-imports|ama|6.0.0
ng-template|Ang|0.1.10
angulardoc-vscode|Ang|6.0.6
vscode-intelephense-client|bme|0.8.8
phpserver|bra|2.4.6
solargraph|cas|0.17.6
npm-intellisense|chr|1.3.0
compulim-vscode-closetag|Com|1.2.0
smoothscroll|cya|1.0.4
angular2-snippets|dan|1.0.20
vscode-material2-snippets|dee|2.0.0
githistory|don|0.4.2
EditorConfig|Edi|0.12.4
tslint|eg2|1.0.39
vsc-material-theme|Equ|2.4.2
php-debug|fel|1.12.5
auto-close-tag|for|0.5.6
vscode-auto-open-markdown-preview|hnw|0.0.4
classic-asp|ili|0.0.4
smarty|imp|0.3.0
mysql-syntax|jak|1.3.1
Angular2|joh|6.1.5
expand-region|let|0.1.2
code-beautifier|mic|2.1.0
vscode-scss|mrm|0.6.2
csharp|ms-|1.16.1
PowerShell|ms-|1.8.4
vsliveshare|ms-|0.3.666
debugger-for-chrome|msj|4.10.1
laravel-blade|one|1.18.0
laravel-extension-pack|one|0.4.0
laravel5-snippets|one|1.5.0
ruby|reb|0.20.0
laravel-artisan|rya|0.0.22
aspnet-helper|sch|0.6.4
code-settings-sync|Sha|3.1.2
nativescript|Tel|0.8.1
nativescript-ng2-snippets|www|2.0.1

</details>

**Comment by two-bridges** [comment]
this is not a duplicate of #58194.  This is a typescript related issue.  It's still occurring.  

I just realised that it is not happening on some of my other TS projects.  So, it's related to this project only, AFAIK.  

I have been coding this project for over a year without any issues.  I was coding on the weekend and it was working fine on Saturday but stopped working on Sunday...

Let me know if you need logs or perhaps I can debug the editor?  Not sure how.  I presume I'd need to load it in Electron or something.

**Comment by mjbvz** [comment]
First does this reproduce in the [latest VS Code insiders build](https://code.visualstudio.com/insiders/)?

You can also run the `Developer: toggle developer tools` command in vscode and look in the dev tools console to see if any additional logs or stack traces are printed there

**Comment by two-bridges** [comment]
Thanks @mjbvz  
I can confirm that the VS Code insiders build works perfectly.  
Also, thanks for the 'toggle developer tools' tip.  I'll try to debug and work out what's happening myself and let you what I find.

**Comment by mjbvz** [comment]
Ok, closing since it sounds like it is fixed in insiders. Let me know if you hit this again

**Comment by two-bridges** [comment]
no worries, thanks for the follow up.  

I coudn't find the cause, but I know it is related to `editorPart.groupViews` has an undefined item in the array.  I haven't determined how the undefined came about or why it only happens in one project.  I have other fish to fry, so I'll stop looking for now.

**Comment by two-bridges** [comment]
@mjbvz I was able to fix the issue.  FYI, it was due to a problem in the Local Storage value: 
storage://workspace/[path to my folder]/memento/workbench.parts.editor

I closed my folder; cleared the LocalStorage value above (using VSCode Developer Tools) and then re-opened the project.  That fixed it.  

Code error:
Within `workbench.parts.editor`, there is a json value `mostRecentActiveGroups`.  This value contained two array items.  However, the `EditorPart.groupViews` member only contained a single item in the array.  This caused `EditorService.findTargetGroup()` to fail on the `group.isOpened()` check.  

I just wanted to document this in case it comes back again...

Cheers
